### PR TITLE
Ensure full sync lock can be broken

### PR DESCRIPTION
**Pull request by gravityrail**
Some small refactors for full sync, and a fix that ensures that full sync can't be permanently blocked by a crash during the enqueuing phase.

This also includes a couple of minor, unrelated cleanups that I discovered while working on it.

### CODE REVIEW

**Comment by lezama** [comment]
LGTM :shipit:

File: json-endpoints/jetpack/class.jetpack-json-api-sync-endpoint.php [hunk 1]
```diff
@@ -5,6 +5,25 @@ class Jetpack_JSON_API_Sync_Endpoint extends Jetpack_JSON_API_Endpoint {
...
+		if ( isset( $args['clear'] ) && $args['clear'] ) {
+			// clear sync queue
+			require_once dirname(__FILE__).'/../../sync/class.jetpack-sync-sender.php';
+
+			$sender = Jetpack_Sync_Sender::getInstance();
+			$sync_queue = $sender->get_sync_queue();
+			$sync_queue->reset();
+		}
+
+		if ( isset( $args['force'] ) && $args['force'] ) {
```
Comment (enejb) [inline json-endpoints/jetpack/class.jetpack-json-api-sync-endpoint.php]: Should we clear the queue everytime we force sync items anyways? 
The only thing we might loose by clearing the queue are events such as login or logout etc.

File: sync/class.jetpack-sync-module-full-sync.php [hunk 2]
```diff
@@ -16,6 +16,7 @@
 
 class Jetpack_Sync_Module_Full_Sync extends Jetpack_Sync_Module {
 	const STATUS_OPTION = 'jetpack_full_sync_status';
+	const FULL_SYNC_TIMEOUT = 3600;
```
Comment (enejb) [inline sync/class.jetpack-sync-module-full-sync.php]: I think it might be good to leave a comment here explaining that its 1 Hour.

**Comment by enejb** [comment]
This is really great. I left a question. But I think this is ready to merge.

File: json-endpoints/jetpack/class.jetpack-json-api-sync-endpoint.php (hunk 1 above)
Comment (lezama) [inline json-endpoints/jetpack/class.jetpack-json-api-sync-endpoint.php]: I don't think we should clear every time, just when things get ugly :)

File: sync/class.jetpack-sync-module-full-sync.php (hunk 2 above)
Comment (lezama) [inline sync/class.jetpack-sync-module-full-sync.php]: Oh good shot, I think we should use WP time constants whenever is posible:

```MINUTE_IN_SECONDS  = 60 (seconds)
HOUR_IN_SECONDS    = 60 * MINUTE_IN_SECONDS
DAY_IN_SECONDS     = 24 * HOUR_IN_SECONDS
WEEK_IN_SECONDS    = 7 * DAY_IN_SECONDS
YEAR_IN_SECONDS    = 365 * DAY_IN_SECONDS```

**Comment by gravityrail** [comment]
Merged to branch-4.2

**Pull request merged**